cron/src/test/java/com/foursquare/service/TrendingCategoriesServiceImplTest.java

```diff
@@ -0,0 +1,190 @@
+package com.foursquare.service;
+
+import com.foursquare.entity.Category;
+import com.foursquare.entity.User;
+import com.foursquare.entity.Venue;
+import com.foursquare.repository.UserRepository;
+import com.foursquare.repository.VenueRepository;
+import com.foursquare.service.impl.TrendingCategoriesServiceImpl;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.mail.javamail.JavaMailSender;
+
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class TrendingCategoriesServiceImplTest {
+
+    @InjectMocks
+    private TrendingCategoriesServiceImpl trendingCategoriesService;
+
+    @Mock
+    private VenueRepository venueRepository;
+    @Mock
+    private UserRepository userRepository;
+    @Mock
+    private JavaMailSender mailSender;
+
+    private Category categoryFirst, categorySecond;
```

Comment:
Some categories are fields, some are variables...it's a mess. This code is not supportable. Extract helper method for category and venue creation.